playlist/forms.py

```diff
@@ -28,29 +42,24 @@ def __init__(self, *args, **kwargs):
 
         return super(SongForm, self).__init__(*args, **kwargs)
 
-    def save(self):
-        """ Song creation needs user and playlist 
-        """
-        song = Song.objects.create(
-            title=self.cleaned_data['title'],
-            link=self.cleaned_data['link'],
-            user=self.user,
-            playlist=self.playlist
-        )
-        return song
-
     def clean_link(self):
         """ check youtube id length
         """
         if len(self.cleaned_data['link']) < 11:
             raise forms.ValidationError("Youtube id length is invalid.")
         return self.cleaned_data['link']
 
-
-class UpdateSongForm(forms.ModelForm):
-    """ Form for editing a song from a playlist
-    """
-
-    class Meta:
-        model = Song
-        fields = ['title', 'link']
\ No newline at end of file
+    def save(self, commit=True):
+        """ Song creation needs user and playlist 
+        """
+        instance = super(SongForm, self).save(commit=False)
+        #checks if self.user and self.playlist exists to determine if it's a creation process
+        if self.user and self.playlist:
```

Comment:
use ternary operator instead of if else statement.
myvar = `foo` if foo else `bar`